packages/flutter/lib/src/material/material.dart

```diff
@@ -305,16 +331,44 @@ class _MaterialState extends State<Material> with TickerProviderStateMixin {
   final GlobalKey _inkFeatureRenderer = GlobalKey(debugLabel: 'ink renderer');
 
   Color _getBackgroundColor(BuildContext context) {
-    if (widget.color != null)
-      return widget.color;
-    switch (widget.type) {
-      case MaterialType.canvas:
-        return Theme.of(context).canvasColor;
-      case MaterialType.card:
-        return Theme.of(context).cardColor;
-      default:
-        return null;
+    Color color = widget.color;
+    if (color == null) {
```

Comment:
Usually we create a temporary theme variable like:
`final ThemeData theme = Theme.of(context);`